src/backend/BSSched/ProjPacks.pm

```diff
@@ -212,6 +212,55 @@ sub get_projpacks {
   return 1;
 }
 
+sub trigger_auto_deep_checks {
+  my ($gctx, $projid, $oldproj, $newproj) = @_;
+  my %badprp;
+  my $oldconfig = $oldproj->{'config'} || '';
+  my $newconfig = $newproj->{'config'} || '';
+  if ($BSConfig::deep_check_dependent_projects_on_macro_change) {
+    for my $oldrepo (@{$oldproj->{'repository'} || []}) {
+      my $repoid = $oldrepo->{'name'};
+      my $newrepo = (grep {$_->{'name'} eq $repoid} @{$oldproj->{'repository'} || []})[0];
+      if (!$newrepo || !BSUtil::identical($oldrepo->{'path'}, $newrepo->{'path'})) {
+	$badprp{"$projid/$repoid"} = 1;
+      } elsif ($oldconfig ne $newconfig) {
+	my @mprefix = ("%define _project $projid", "%define _repository $repoid");
+	my $cold = Build::read_config($gctx->{'arch'}, [ @mprefix, split("\n", $oldconfig) ]);
+	my $cnew = Build::read_config($gctx->{'arch'}, [ @mprefix, split("\n", $newconfig) ]);
+	$badprp{"$projid/$repoid"} = 1 if !BSUtil::identical($cold->{'macros'}, $cnew->{'macros'});
+      }
+    }
+  } elsif ($oldconfig ne $newconfig) {
+    my @mprefix = ("%define _project $projid");
+    my $cold = Build::read_config($gctx->{'arch'}, [ @mprefix, split("\n", $oldproj->{'config'} || '') ]);
+    my $cnew = Build::read_config($gctx->{'arch'}, [ @mprefix, split("\n", $newproj->{'config'} || '') ]);
+    if (($cold->{'expandflags:macroserial'} || '') ne ($cold->{'expandflags:macroserial'} || '')) {
```

Comment:
That was likely meant to compare cold to cnew; otherwise it should never be 'ne' (and this likely explains why the feature is not working as intended in OBS)